source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/ActionLayoutRenderer.java

```diff
@@ -0,0 +1,126 @@
+import io.adaptivecards.renderer.IActionLayoutRenderer;
+import io.adaptivecards.renderer.IBaseActionElementRenderer;
+import io.adaptivecards.renderer.RenderedAdaptiveCard;
+import io.adaptivecards.renderer.actionhandler.ICardActionHandler;
+import io.adaptivecards.renderer.registration.CardRendererRegistration;
+
+public class ActionLayoutRenderer implements IActionLayoutRenderer {
+
+    protected ActionLayoutRenderer()
+    {
+    }
+
+    public static ActionLayoutRenderer getInstance()
+    {
+        if (s_instance == null)
+        {
+            s_instance = new ActionLayoutRenderer();
+        }
+
+        return s_instance;
+    }
+
+    public void renderActions(RenderedAdaptiveCard renderedCard, Context context, FragmentManager fragmentManager, ViewGroup viewGroup, BaseActionElementVector baseActionElementList, ICardActionHandler cardActionHandler, HostConfig hostConfig) {
+        long size;
+        if (baseActionElementList == null || (size = baseActionElementList.size()) <= 0)
+        {
+            return;
+        }
+
+        LinearLayout actionButtonsLayout = new LinearLayout(context);
+        actionButtonsLayout.setLayoutParams(new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));
+        int alignment = hostConfig.getActions().getActionAlignment().swigValue();
+        if (alignment == ActionAlignment.Right.swigValue())
+        {
+            actionButtonsLayout.setGravity(Gravity.RIGHT);
+        }
+        else if (alignment == ActionAlignment.Center.swigValue())
+        {
+            actionButtonsLayout.setGravity(Gravity.CENTER_HORIZONTAL);
+        }
+
+        int actionButtonsLayoutOrientation = hostConfig.getActions().getActionsOrientation().swigValue();
+        if (actionButtonsLayoutOrientation == ActionsOrientation.Vertical.swigValue())
+        {
+            actionButtonsLayout.setOrientation(LinearLayout.VERTICAL);
+        }
+        else
+        {
+            actionButtonsLayout.setOrientation(LinearLayout.HORIZONTAL);
+        }
+
+
+        Spacing spacing = hostConfig.getActions().getSpacing();
+        /* Passing false for separator since we do not have any configuration for separator in actionsConfig */
+        BaseCardElementRenderer.setSpacingAndSeparator(context, viewGroup, spacing, false, hostConfig, true /* Horizontal Line */);
+
+        if (viewGroup != null)
+        {
+            if(actionButtonsLayoutOrientation == ActionsOrientation.Horizontal.swigValue())
+            {
+                HorizontalScrollView actionButtonsContainer = new HorizontalScrollView(context);
+                actionButtonsContainer.setHorizontalScrollBarEnabled(false);
+                actionButtonsContainer.addView(actionButtonsLayout);
+                viewGroup.addView(actionButtonsContainer);
+            }
+            else
+            {
+                viewGroup.addView(actionButtonsLayout);
+            }
+        }
+
+        int i = 0;
```

Comment:
any reason why we wouldn't want to just declare `i` independently in each `for` loop below?